File: src/main/java/ir/reservs/reservs/di/component/ActivityComponent.java
package ir.reservs.reservs.di.component;

import dagger.Component;
import ir.reservs.reservs.di.PerActivity;
import ir.reservs.reservs.di.module.ActivityModule;
import ir.reservs.reservs.ui.login.LoginActivity;
import ir.reservs.reservs.ui.main.MainActivity;
import ir.reservs.reservs.ui.splash.SplashActivity;


@PerActivity
@Component(dependencies = ApplicationComponent.class, modules = ActivityModule.class)
public interface ActivityComponent {
    void inject(SplashActivity activity);

    void inject(LoginActivity activity);

    void inject(MainActivity activity);


}

File: src/main/java/ir/reservs/reservs/di/component/ActivityComponent.kt
package ir.reservs.reservs.di.component;

import dagger.Component
import ir.reservs.reservs.di.PerActivity
import ir.reservs.reservs.di.module.ActivityModule
import ir.reservs.reservs.ui.main.MainActivity


@PerActivity
@Component(dependencies = [ApplicationComponent::class], modules = [ActivityModule::class])
interface ActivityComponent {

    fun inject(activity: MainActivity)

}

